packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+import * as vscode from 'vscode';
+import crypto = require('crypto');
+import events = require('events');
+import fs = require('fs');
+import ospath = require('path');
+import {
+  CliCommandExecutor,
+  Command
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { ApexTestRequestInfo, getApexTests, isLanguageClientReady } from '..';
+import { nls } from '../messages';
+
+// Import from core
+const sfdxCoreExports = vscode.extensions.getExtension(
+  'salesforce.salesforcedx-vscode-core'
+)!.exports;
+const ProgressNotification = sfdxCoreExports.ProgressNotification;
+const EmptyParametersGatherer = sfdxCoreExports.EmptyParametersGatherer;
+const SfdxCommandlet = sfdxCoreExports.SfdxCommandlet;
+const ForceApexTestRunCodeActionExecutor =
+  sfdxCoreExports.ForceApexTestRunCodeActionExecutor;
+const SfdxWorkspaceChecker = sfdxCoreExports.SfdxWorkspaceChecker;
+const channelService = sfdxCoreExports.channelService;
+const notificationService = sfdxCoreExports.notificationService;
+const taskViewService = sfdxCoreExports.taskViewService;
+
+const LIGHT_BLUE_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testNotRun.svg'
+);
+const LIGHT_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testFail.svg'
+);
+const LIGHT_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testPass.svg'
+);
+
+const DARK_BLUE_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testNotRun.svg'
+);
+const DARK_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testFail.svg'
+);
+const DARK_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testPass.svg'
+);
+
+export type FullTestResult = {
+  summary: TestSummary;
+  tests: TestResult[];
+};
+
+export type TestSummary = {
+  outcome: string;
+  testsRan: number;
+  passing: number;
```

Comment:
That's just what it is called in the actual json output, so I left that as that

Reply:
OK. Maybe we should move all these files that are just mapping the JSON into a separate file called TestDataAccessObjects.ts so that it's clearer that this is just mapping to the JSON since you have no control over that.

The JSON is "stable" right? And won't change?

Reply:
It shouldn't change unless there is a change to the actual sfdx force:apex cli commands themselves